### PR TITLE
Fix 'verifyObject' for class objects with staticmethods on Python 3.

### LINKED ISSUE #126: BrokenMethodImplementation "doesn't allow enough arguments" error when using staticmethods on Python 3
Verifying a class using `@provider` behaves differently on Python 2.7 vs. Python 3.7 with zope.interface 4.5.0:

```python
import zope.interface
from zope.interface.verify import verifyObject

class IFoo(zope.interface.Interface):
    def bar(a, b):
        pass

@zope.interface.provider(IFoo)
class Foo(object):
    @staticmethod
    def bar(a, b):
        pass

assert verifyObject(IFoo, Foo)
```

On Python 2.7, this code passes. On Python 3.7 it raises a `BrokenMethodImplementation`:

```pytb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/andy/test/.tox/py37/lib/python3.7/site-packages/zope/interface/verify.py", line 105, in verifyObject
    return _verify(iface, candidate, tentative, vtype='o')
  File "/Users/andy/test/.tox/py37/lib/python3.7/site-packages/zope/interface/verify.py", line 97, in _verify
    raise BrokenMethodImplementation(name, mess)
zope.interface.exceptions.BrokenMethodImplementation: The implementation of bar violates its contract
        because implementation doesn't allow enough arguments.
```

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #126

### CODE REVIEW

File: src/zope/interface/tests/test_verify.py [hunk 1]
```diff
@@ -557,5 +557,26 @@ class IDummyModule(Interface):
...
 
+    def test_staticmethod_hit_on_class(self):
+        from zope.interface import Interface
+        from zope.interface import provider
+        from zope.interface.verify import verifyObject
+
+        class IFoo(Interface):
+
+            def bar(a, b):
+                pass
```
Comment (rouge8) [inline src/zope/interface/tests/test_verify.py]: I think to make coverage happy you can change this to `raise NotImplementedError()`

Comment (jamadden) [reply src/zope/interface/tests/test_verify.py]: Whoops, usually I'm pretty good about that kind of thing. Thanks!

**Comment by jamadden** [comment]
Thanks! 4.6.0 has been released to PyPI.

**Comment by rouge8** [comment]
Thanks for the fast fix!

**Pull request merged**